### PR TITLE
[YUNIKORN-2541] Upgrade golang.org/x/net for CVEs

### PR DESCRIPTION
**Pull request by targetoee**
### What is this PR for?
This PR upgrades `golang.org/x/net` to 0.23.0 to address CVEs.
Additionally, it updates the versions of the scheduler-interface and core, as they also use this module and have been updated accordingly.

### What type of PR is it?
* [ ] - Bug Fix
* [x] - Improvement
* [ ] - Feature
* [ ] - Documentation
* [ ] - Hot Fix
* [ ] - Refactoring

### Todos
N/A

### What is the Jira issue?
[YUNIKORN-2541](https://issues.apache.org/jira/browse/YUNIKORN-2541/)

### How should this be tested?
It's tested by `make test` locally and all tests passed.

### Screenshots (if appropriate)
N/A

### Questions:
N/A

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -122,13 +122,13 @@ require (
 	go.opentelemetry.io/proto/otlp v1.0.0 // indirect
 	go.starlark.net v0.0.0-20230525235612-a134d8f9ddca // indirect
 	go.uber.org/multierr v1.11.0 // indirect
-	golang.org/x/crypto v0.19.0 // indirect
+	golang.org/x/crypto v0.21.0 // indirect
```
Comment (chenyulin0719) [inline go.mod:125]:
```
golang.org/x/crypto v0.21.0 // indirect
golang.org/x/sys v0.18.0 // indirect
golang.org/x/term v0.18.0 // indirect
```

Above dependencies are automatically upgrade by 'go mod tidy' after upgared golang.org/x/net to v0.23.0.

File: go.mod [hunk 2]
```diff
@@ -164,10 +164,10 @@ require (
 replace (
 	github.com/opencontainers/runc => github.com/opencontainers/runc v1.1.12
 	github.com/petermattis/goid => github.com/petermattis/goid v0.0.0-20240327183114-c42a807a84ba
-	golang.org/x/crypto => golang.org/x/crypto v0.19.0
+	golang.org/x/crypto => golang.org/x/crypto v0.21.0
```
Comment (chenyulin0719) [inline go.mod:167]: Since indirect require have a newer version, so @targetoee update replace golang.org/x/crypto to v0.21.0 as  I suggested.

Comment (chenyulin0719) [reply go.mod:167]: @wilfred-s, @craigcondit , @chia7712, 
I remember that we have concluded we should keep all the replace directive aligned across all the repositories. So I will request @targetoee to update replcae directive golang.org/x/crypto to v0.21.0 in core and shceduler-interface as well.

**Review by chenyulin0719 — APPROVED** [review]
LGTM, we should algin the replace directive change in core and scheduler-interface. Please sent ADDENDUM PR once we got a confirmation from @wilfred-s or @craigcondit .

File: go.mod (hunk 2 above)
Comment (targetoee) [reply go.mod:167]: Thanks for the reminder. I've updated the scheduler-interface first (https://github.com/apache/yunikorn-scheduler-interface/pull/136) and will update the core after.

**Review by wilfred-s — CHANGES_REQUESTED** [review]
should wait for the SI change, and also check the core for an update to the SI and the crypto dependency.

File: go.mod (hunk 2 above)
Comment (chenyulin0719) [reply go.mod:167]: @targetoee Shim and core are updated, please update this PR with the latest version.

**Review by chenyulin0719 — APPROVED** [review]
LGTM, waiting for e2e test.

**Comment by targetoee** [comment]
The crypto dependency has been updated in both the SI and core projects, and k8shim has also been updated accordingly.

**Pull request closed without merging**